NEW: Add support for Silverstripe 5

mikenuguid opened the pull request.

mikenuguid committed:
NEW: Add support for Silverstripe 5

jonom commented on README.md:
```diff
@@ -18,7 +18,7 @@ Say you have a contact form on every page of your site, and you have a `ModelAdm
 ## Requirements
 
-SilverStripe ^4
+SilverStripe ^5
```
```suggestion
SilverStripe ^4 | ^5
```

jonom reviewed (commented):
Thanks @mikenuguid, did you test that everything works for SS5?

mikenuguid commented:
> Thanks @mikenuguid, did you test that everything works for SS5?

Thanks for the feedback @jonom. Just a small kink on this: https://github.com/jonom/silverstripe-someconfig/blob/master/src/SomeConfigAdmin.php#L21

Since we're using generators for Datalist (and subclasses, including ArrayList) iterating by reference will not work properly. We could remove the reference to fix it. Please let me know if I should create an issue for it or I could add the fix to this PR. 🙂 

Reference: https://docs.silverstripe.org/en/5/changelogs/5.0.0/?_ga=2.128522828.1953607834.1685309850-1465694130.1681331049#api-orm

jonom commented:
Interesting! I'm not sure why I had the reference there... it shouldn't be necessary should it? Yes feel free to add to this PR. Thanks.

mikenuguid committed:
FIX Remove reference on datalist

- Datalist and subclasses e.g. ArrayList now utilises generators which creates issues when iterated by reference
- Minor tidy-up on `ReadMe.md`

jonom commented:
Sorry for the delay @mikenuguid and thanks! I've tagged 1.0.3 with this change.
